File: app/src/main/java/me/wcy/music/activity/PlaylistActivity.java
package me.wcy.music.activity;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AlertDialog;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

import me.wcy.music.R;
import me.wcy.music.adapter.OnMoreClickListener;
import me.wcy.music.adapter.PlaylistAdapter;
import me.wcy.music.model.Music;
import me.wcy.music.service.AudioPlayer;
import me.wcy.music.service.OnPlayerEventListener;
import me.wcy.music.utils.binding.Bind;

/**
 * 播放列表
 */
public class PlaylistActivity extends BaseActivity implements AdapterView.OnItemClickListener, OnMoreClickListener, OnPlayerEventListener {
    @Bind(R.id.lv_playlist)
    private ListView lvPlaylist;

    private PlaylistAdapter adapter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_playlist);
    }

    @Override
    protected void onServiceBound() {
        adapter = new PlaylistAdapter(AudioPlayer.get().getMusicList());
        adapter.setIsPlaylist(true);
        adapter.setOnMoreClickListener(this);
        lvPlaylist.setAdapter(adapter);
        lvPlaylist.setOnItemClickListener(this);
        AudioPlayer.get().addOnPlayEventListener(this);
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
        AudioPlayer.get().play(position);
    }

    @Override
    public void onMoreClick(int position) {
        String[] items = new String[]{"移除"};
        Music music = AudioPlayer.get().getMusicList().get(position);
        AlertDialog.Builder dialog = new AlertDialog.Builder(this);
        dialog.setTitle(music.getTitle());
        dialog.setItems(items, (dialog1, which) -> {
            AudioPlayer.get().delete(position);
            adapter.notifyDataSetChanged();
        });
        dialog.show();
    }

    @Override
    public void onChange(Music music) {
        adapter.notifyDataSetChanged();
    }

    @Override
    public void onPlayerStart() {
    }

    @Override
    public void onPlayerPause() {
    }

    @Override
    public void onPublish(int progress) {
    }

    @Override
    public void onBufferingUpdate(int percent) {
    }

    @Override
    protected void onDestroy() {
        AudioPlayer.get().removeOnPlayEventListener(this);
        super.onDestroy();
    }
}

File: app/src/main/java/me/wcy/music/activity/PlaylistActivity.kt
package me.wcy.music.activity

import android.content.DialogInterface
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.ListView
import androidx.appcompat.app.AlertDialog
import me.wcy.music.R
import me.wcy.music.adapter.OnMoreClickListener
import me.wcy.music.adapter.PlaylistAdapter
import me.wcy.music.model.Music
import me.wcy.music.service.AudioPlayer
import me.wcy.music.service.OnPlayerEventListener
import me.wcy.music.utils.binding.Bind

/**
 * 播放列表
 */
class PlaylistActivity : BaseActivity(), OnItemClickListener, OnMoreClickListener,
    OnPlayerEventListener {
    @Bind(R.id.lv_playlist)
    private val lvPlaylist: ListView? = null
    private var adapter: PlaylistAdapter? = null
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_playlist)
        adapter = PlaylistAdapter(AudioPlayer.get().getMusicList())
        adapter!!.setIsPlaylist(true)
        adapter!!.setOnMoreClickListener(this)
        lvPlaylist!!.adapter = adapter
        lvPlaylist.onItemClickListener = this
        AudioPlayer.get().addOnPlayEventListener(this)
    }

    override fun onItemClick(parent: AdapterView<*>?, view: View, position: Int, id: Long) {
        AudioPlayer.get().play(position)
    }

    override fun onMoreClick(position: Int) {
        val items = arrayOf("移除")
        val music: Music = AudioPlayer.get().getMusicList()!!.get(position)
        val dialog = AlertDialog.Builder(this)
        dialog.setTitle(music.title)
        dialog.setItems(items) { dialog1: DialogInterface?, which: Int ->
            AudioPlayer.get().delete(position)
            adapter!!.notifyDataSetChanged()
        }
        dialog.show()
    }

    override fun onChange(music: Music?) {
        adapter!!.notifyDataSetChanged()
    }

    override fun onPlayerStart() {}
    override fun onPlayerPause() {}
    override fun onPublish(progress: Int) {}
    override fun onBufferingUpdate(percent: Int) {}
    override fun onDestroy() {
        AudioPlayer.get().removeOnPlayEventListener(this)
        super.onDestroy()
    }
}